### PR TITLE
Clean up files for deletion to save disk space

### PR DESCRIPTION
**Pull request by rekhaswaminathan**

### CODE REVIEW

File: src/Diagnostics.RuntimeHost/Services/SourceWatcher/Watchers/GitHubWatcher.cs [hunk 1]
```diff
@@ -418,5 +419,55 @@ private void LoadConfigurations()
                 Directory.CreateDirectory(_destinationCsxPath);
             }
         }       
+
+        private async void CleanupFilesForDeletion()
```
Comment (ShekharGupta1988) [inline src/Diagnostics.RuntimeHost/Services/SourceWatcher/Watchers/GitHubWatcher.cs:423]: async void is not recommended as the best practice for async programming. 
Change it to return Task.

There are lot of articles explaining this. You can also read it here : https://docs.microsoft.com/en-us/archive/msdn-magazine/2013/march/async-await-best-practices-in-asynchronous-programming

File: src/Diagnostics.RuntimeHost/Services/SourceWatcher/Watchers/GitHubWatcher.cs [hunk 2]
```diff
@@ -418,5 +419,55 @@ private void LoadConfigurations()
...
+                    }
+                }
+            }
+            catch (Exception ex)
+            {
+                LogException($"CleanupDataFilesMarkedForDeletion Failed. Exception : {ex.ToString()}", ex);
+            }
+        }
+
+        private void DeleteFolderRecursive(DirectoryInfo baseDir)
```
Comment (ShekharGupta1988) [inline src/Diagnostics.RuntimeHost/Services/SourceWatcher/Watchers/GitHubWatcher.cs:461]: Consider moving this to FileHelper utility class.

File: src/Diagnostics.RuntimeHost/Services/SourceWatcher/Watchers/GitHubWatcher.cs [hunk 3]
```diff
@@ -418,5 +419,55 @@ private void LoadConfigurations()
...
+        {
+            baseDir.Attributes = FileAttributes.Normal;
+            foreach (var childDir in baseDir.GetDirectories())
+                DeleteFolderRecursive(childDir);
+
+            foreach (var file in baseDir.GetFiles())
+                file.IsReadOnly = false;
+
+            baseDir.Delete(true);
```
Comment (ShekharGupta1988) [inline src/Diagnostics.RuntimeHost/Services/SourceWatcher/Watchers/GitHubWatcher.cs:470]: Just check once but I think this call is sync in nature and will block the thread. Generally, Basic IO File operations are sync unless you use FileStream methods.

File: src/Diagnostics.RuntimeHost/Services/SourceWatcher/Watchers/GitHubWatcher.cs [hunk 4]
```diff
@@ -418,5 +419,55 @@ private void LoadConfigurations()
                 Directory.CreateDirectory(_destinationCsxPath);
             }
         }       
+
+        private async void CleanupFilesForDeletion()
+        {
+            try
+            {
+                DirectoryInfo scriptsDir = new DirectoryInfo(_destinationCsxPath);
```
Comment (ShekharGupta1988) [inline src/Diagnostics.RuntimeHost/Services/SourceWatcher/Watchers/GitHubWatcher.cs:427]: Can you log the start and end of this cleanup method

**Comment by hforeste** [comment]
Hey @rekhaswaminathan , what is the context of this change?

**Comment by rekhaswaminathan** [comment]
> Hey @rekhaswaminathan , what is the context of this change?

The context is size of bin/Detectors folders is growing and sometimes local cache initialization is failing during swap. So I wanted to implement some clean up mechanism to save disk space.

File: src/Diagnostics.RuntimeHost/Services/SourceWatcher/Watchers/GitHubWatcher.cs [hunk 5]
```diff
@@ -418,5 +421,57 @@ private void LoadConfigurations()
                 Directory.CreateDirectory(_destinationCsxPath);
             }
         }       
+
+        private async void StartCleanupTask()
+        {
```
Comment (ShekharGupta1988) [inline src/Diagnostics.RuntimeHost/Services/SourceWatcher/Watchers/GitHubWatcher.cs]: just get rid of this method. Rest all looks good.

**Pull request merged**